clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoAdminClient.java

```diff
@@ -0,0 +1,239 @@
+    NameIdentifier.checkMetalake(ident);
+
+    MetalakeCreateRequest req = new MetalakeCreateRequest(ident.name(), comment, properties);
+    req.validate();
+
+    MetalakeResponse resp =
+        restClient.post(
+            API_METALAKES_LIST_PATH,
+            req,
+            MetalakeResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.metalakeErrorHandler());
+    resp.validate();
+
+    return DTOConverters.toMetaLake(resp.getMetalake(), restClient);
+  }
+
+  /**
+   * Alters a specific Metalake using the Gravitino API.
+   *
+   * @param ident The identifier of the Metalake to be altered.
+   * @param changes The changes to be applied to the Metalake.
+   * @return A GravitinoMetaLake instance representing the updated Metalake.
+   * @throws NoSuchMetalakeException If the specified Metalake does not exist.
+   * @throws IllegalArgumentException If the provided changes are invalid or not applicable.
+   */
+  @Override
+  public GravitinoMetaLake alterMetalake(NameIdentifier ident, MetalakeChange... changes)
+      throws NoSuchMetalakeException, IllegalArgumentException {
+    NameIdentifier.checkMetalake(ident);
+
+    List<MetalakeUpdateRequest> reqs =
+        Arrays.stream(changes)
+            .map(DTOConverters::toMetalakeUpdateRequest)
+            .collect(Collectors.toList());
+    MetalakeUpdatesRequest updatesRequest = new MetalakeUpdatesRequest(reqs);
+    updatesRequest.validate();
+
+    MetalakeResponse resp =
+        restClient.put(
+            API_METALAKES_IDENTIFIER_PATH + ident.name(),
+            updatesRequest,
+            MetalakeResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.metalakeErrorHandler());
+    resp.validate();
+
+    return DTOConverters.toMetaLake(resp.getMetalake(), restClient);
+  }
+
+  /**
+   * Drops a specific Metalake using the Gravitino API.
+   *
+   * @param ident The identifier of the Metalake to be dropped.
+   * @return True if the Metalake was successfully dropped, false otherwise.
+   */
+  @Override
+  public boolean dropMetalake(NameIdentifier ident) {
+    NameIdentifier.checkMetalake(ident);
+
+    try {
+      DropResponse resp =
+          restClient.delete(
+              API_METALAKES_IDENTIFIER_PATH + ident.name(),
+              DropResponse.class,
+              Collections.emptyMap(),
+              ErrorHandlers.metalakeErrorHandler());
+      resp.validate();
+      return resp.dropped();
+
+    } catch (Exception e) {
+      LOG.warn("Failed to drop metadata {}", ident, e);
+      return false;
+    }
+  }
+
+  /**
+   * Creates a new builder for constructing a GravitinoClient.
+   *
+   * @param uri The base URI for the Gravitino API.
+   * @return A new instance of the Builder class for constructing a GravitinoClient.
+   */
+  public static Builder builder(String uri) {
+    return new Builder(uri);
+  }
+
+  /** Builder class for constructing a GravitinoClient. */
+  public static class Builder {
```

Comment:
Maybe we can also have a base builder class, since some of the methods are the same.

Reply:
> Maybe we can also have a base builder class, since some of the methods are the same.

OK

Reply:
Yes, let me do that.

Reply:
@jerryshao this has been improved as you suggested; the code is much clean now. Please review, thx!